src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String[] data = readFile(fromFileName);
+            String report = createReport(data);
+            writeToFile(report, toFileName);
+            System.out.println("Statistics generated and written to " + toFileName);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read or write file", e);
+        }
+    }
+
+    private String[] readFile(String fromFileName) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            return reader.lines().toArray(String[]::new);
+        }
+    }
+
+    private String createReport(String[] data) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+        StringBuilder reportBuilder = new StringBuilder();
+
+        for (String line : data) {
```

Comment:
This logic is appropriate to readFile method, where you read and collect data to int[] array for example